server.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+import argparse
+import glob
+import json
+import logging
+import os
+import sys
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from pathlib import Path
+
+import yaml
+from jinja2 import Template
+
+TARGET_NAMESPACE_LABEL = os.getenv("LABEL", "user.kubeflow.org/enabled")
+TEMPLATES_FOLDER = os.getenv("TEMPLATES_FOLDER", "./resources")
+PORT = os.getenv("PORT", 80)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+handler = logging.StreamHandler(sys.stdout)
+handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+def main(port: int, label: str, folder: str) -> None:
+    """
+    Runs a server for injecting resources into namespaces by given label
+    """
+    logger.info("Resource dyspatcher service alive")
+    server = server_factory(port, label, folder)
+    logger.info("Serving forever")
+    server.serve_forever()
+
+
+def server_factory(controller_port: int, label: str, folder: str, url: str = "") -> HTTPServer:
+    """
+    Returns an HTTPServer populated with Handler with customized settings
```

Comment:
nit: Canonical uses British english
```suggestion
    Returns an HTTPServer populated with Handler with customised settings
```